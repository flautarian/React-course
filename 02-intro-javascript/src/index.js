console.log("Hello world!!!")

const dice = {
    faces: 20,
    getRandomValue: function(){
        // 👇️ get number between min (inclusive) and max (inclusive)
        return Math.floor(Math.random() * (this.faces+ 1));
    }
}

const diceResult =`${dice.getRandomValue()}`;
console.log(diceResult);