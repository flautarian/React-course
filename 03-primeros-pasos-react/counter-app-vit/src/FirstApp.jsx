import React from 'react'
import { Fragment } from 'react';

function getPerson() {
    let name = prompt("cual es tu nombre?") || "leonidas";
    let surname = prompt("cual es tu apellido?") || "deuteronomus";
    let job = prompt("cual es tu oficio?") || "espartano";
    return {
        name: name,
        surname: surname,
        job: job
    };
}

let person = getPerson();

export const FirstApp = () => {
    return (
        <Fragment>
            <p>Nombre:</p>
            <p>-Mi nombre es {person.name}</p>
            <p>-Mi apellido es {person.surname}</p>
            <p>-Y mi trabajo es {person.job}</p>
        </Fragment>
    )
}
