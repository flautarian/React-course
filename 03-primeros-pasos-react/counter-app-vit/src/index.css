html, body {
    background-color: #21232A;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1em;
    padding: 70px;
  }
  
  code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
      monospace;
  }
  
button {
  padding: 15px;
  margin: 5px;
  font-size: 1.3em;
}