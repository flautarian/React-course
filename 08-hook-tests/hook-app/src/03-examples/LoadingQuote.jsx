import React, { Fragment } from 'react'

export const LoadingQuote = () => {

    return (
        <Fragment>
            <div className='alert alert-info text-center'>
                Loading...
            </div>
        </Fragment>
    )

}
