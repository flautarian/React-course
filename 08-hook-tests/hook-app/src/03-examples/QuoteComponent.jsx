import React, { Fragment, useEffect, useRef } from 'react'
import { useState } from 'react';
import { useLayoutEffect } from 'react'

export const QuoteComponent = ({ data }) => {

    const pRef = useRef();
    const [boxSize, setBoxSize] = useState({ width: 0, height: 0 });

    useLayoutEffect(() => {
        const { height, width } = pRef.current.getBoundingClientRect();
        setBoxSize({ height, width });
    }, [data])

    return (
        <Fragment>
            <blockquote  className='blockquote text-start' style={{ display: 'flex' }}>
                <p ref={pRef} className='mb-1'>{data.quote} - {data.author}</p>

                <footer className='blockquote-footer'> footer </footer>
            </blockquote>
            <code>{ JSON.stringify(boxSize) }</code>
            <hr/>
        </Fragment>
    )
}
