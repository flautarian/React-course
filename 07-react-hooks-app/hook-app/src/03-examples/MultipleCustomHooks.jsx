import React, { Fragment, useEffect } from 'react'
import { CounterApp } from '../01-useState/CounterApp';
import { useCounter, useFetch } from '../hooks/index';
import { LoadingQuote } from './LoadingQuote';
import { QuoteComponent } from './QuoteComponent';

export const MultipleCustomHooks = () => {

    const { valueCounter, changeValue } = useCounter(1);

    const { data, isLoading, hasError } = useFetch(`https://breakingbadapi.com/api/quotes/${valueCounter.counter2}`);

    return (
        <Fragment>
            <h1>Breaking bad Quotes</h1>
            <hr />
            {
                isLoading ?
                    <LoadingQuote /> :
                    <QuoteComponent data={data[0]} />
            }

            <button className='btn btn-primary' onClick={() => changeValue([0, 1, 0])}>Next quote</button>
        </Fragment>
    )
}
