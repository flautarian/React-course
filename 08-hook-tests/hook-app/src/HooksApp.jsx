import { Fragment } from "react"
import { CounterApp } from "./01-useState/CounterApp"
import { CounterAppWithCustomHook } from "./01-useState/CounterAppWithCustomHook"
import { SimpleForm } from "./02-useEffect/SimpleForm"
import { SimpleFormWithCustomHook } from "./02-useEffect/SimpleFormWithCustomHook"
import { MultipleCustomHooks } from "./03-examples/MultipleCustomHooks"
import { FocusScreen } from "./04-useRef/FocusScreen"
import { Layout } from "./05-useLayoutEffect/Layout"
import { CallbackHook } from "./06-memos/CallbackHook"
import { MemoHook } from "./06-memos/MemoHook"
import { Memorize } from "./06-memos/Memorize"
import { Padre } from "./07-tarea-memo/Padre"
import { TodoApp } from "./08-use-reducer/TodoApp"
import { MainApp } from "./09-use-context/MainApp"

export const HooksApp = () => {
    return (
        <Fragment>
            <h2>Hooks App</h2>
            {/* <CounterApp value={0}/> */}
            {/* <CounterAppWithCustomHook value={0}/> */}
            {/* <SimpleForm/> */}
            {/* <SimpleFormWithCustomHook/> */}
            {/* <MultipleCustomHooks/> */}
            {/* <FocusScreen/> */}
            {/* <Layout/> */}
            {/* <Memorize/> */}
            {/* <MemoHook/> */}
            {/* <CallbackHook/> */}
            {/* <Padre/> */}
            {/* <TodoApp/> */}
            <MainApp/>
        </Fragment>
    )
}