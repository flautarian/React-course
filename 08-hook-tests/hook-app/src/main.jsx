import React from 'react';
import { HooksApp } from './HooksApp';
import ReactDOM from 'react-dom/client';
import './index.css';
import { BrowserRouter } from "react-router-dom";

ReactDOM.createRoot(document.getElementById('root')).render(
  <BrowserRouter>
    {/* <React.StrictMode> */}
      <HooksApp />
    {/* </React.StrictMode> */}
  </BrowserRouter>
)
