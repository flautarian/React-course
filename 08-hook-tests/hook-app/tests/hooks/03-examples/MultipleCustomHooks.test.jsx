import { fireEvent, render, screen } from '@testing-library/react';
import { MultipleCustomHooks } from '../../../src/03-examples/MultipleCustomHooks';
import { useCounter } from '../../../src/hooks';
import { useFetch } from '../../../src/hooks/useFetch';

jest.mock('../../../src/hooks/useFetch');
jest.mock('../../../src/hooks/useCounter');

describe('Breaking bad test', () => {

    const mockChangeValue = jest.fn();

    useCounter.mockReturnValue({
        valueCounter: {
            counter1: 1,
            counter2: 1,
            counter3: 1
        },
        changeValue: mockChangeValue,
    });

    useFetch.mockReturnValue({
        data: null,
        isLoading: true,
        hasError: null,
    });

    beforeEach( () => {
        jest.clearAllMocks();
    });

    test('should match snapshot', () => {
        const { container } = render(<MultipleCustomHooks />);
        expect(container).toMatchSnapshot();
    });

    test('should default correctly', () => {
        render(<MultipleCustomHooks />);
        expect(screen.getByText('Loading...'));
        expect(screen.getByText('Next quote'));

        const nextButton = screen.getByRole('button', { name: 'Next quote' });
        expect(nextButton.disabled).toBeFalsy();
    });

    test('should fetch correctly', () => {

        useFetch.mockReturnValue({
            data: [{
                id: "1",
                author: "Facundo",
                quote: 'Hello world',
            }],
            isLoading: false,
            hasError: null,
        });

        render(<MultipleCustomHooks />);
        expect(screen.getByText('Hello world - Facundo')).toBeTruthy();
    });

    test('should fetch next quote correctly', () => {

        useFetch.mockReturnValue({
            data: [{ id: "1", author: "Facundo", quote: 'Hello world' }],
            isLoading: false,
            hasError: null,
        });

        render(<MultipleCustomHooks />);
        expect(screen.getByText('Hello world - Facundo')).toBeTruthy();
        const nextButton = screen.getByRole('button', { name: 'Next quote' });
        fireEvent.click(nextButton);
        expect(mockChangeValue).toHaveBeenCalled();
    });
})