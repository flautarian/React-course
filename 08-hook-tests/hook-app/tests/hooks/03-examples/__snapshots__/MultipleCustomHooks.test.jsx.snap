// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breaking bad test should match snapshot 1`] = `
<div>
  <h1>
    Breaking bad Quotes
  </h1>
  <hr />
  <div
    class="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    class="btn btn-primary"
  >
    Next quote
  </button>
</div>
`;
